public/app.js

```diff
@@ -56,11 +80,17 @@ class Controller {
         Controller.setupEventListeners();
         // Add the dropdown menu to the UI
         DataController.watchActiveProjects(UIController.addDropdownMenu);
-        // Add the coding progress section to the UI
-        UIController.addCodingProgressSection();
-        // Get data for coding progress table
-        let unsubscribeFunc = DataController.watchCodingProgress(UIController.updateProgressUI);
-        DataController.registerSnapshotListener(unsubscribeFunc);
+        // Get hash value
+        Controller.hash = window.location.hash.substring(1);
```

Comment:
I find Controller.hash a confusing name, because I'd expect that to be the hash of the Controller object, not the hash component of the url. Suggest something like e.g. url_hash_component, or page_route